### ISSUE TITLE
lab 3 output sequence

### ISSUE THREAD

**Issue by zzhangyue** [nus-cs2030/2223-s2#459]
would really appreciate some advice on this :")

basically my output is all over the place, for example there's a serve before an arrive ???? is it a problem with the comparator or simulator? thank you!!
<img width="462" alt="Screenshot 2023-03-19 at 15 13 0
<img width="705" alt="Screenshot 2023-03-19 at 15 11 33" src="https://user-images.githubusercontent.com/122248591/226160055-e026dfb1-0ec3-4fd7-9626-a980b403a83d.png">
7" src="https://user-images.githubusercontent.com/122248591/226160028-10ac5a3b-67a4-4d43-84f2-dfafe229e478.png">

**Comment by littlemidget3** [comment]
yeah likely an issue with your comparator. Ensure youre checking both time and cust id

**Comment by dlyxxx** [comment]
you should check that your comparator checks the time that the event is occurring, the earlier event will be the priority, as for those with the same timings, you should check which customer has the lowest customer ID. hope that this helps

**Comment by ayevxn** [comment]
Right now it seems like your comparator is working though, since it's evaluating the events by time (earlier timing executed first). It might be an issue with how your individual events are being executed - what's leading to a done event for 3 and 4 to be initialised at 1.116/1.131?

**Comment by kelleyhan25** [comment]
I think it might be the comparator, probably with customer ID if you aren't checking that

**Comment by dmidula** [comment]
Might be that you are not returning the correct event times for your Done Event as your events still seem to be in chronological order, meaning there probably isn't a problem with your comparator.

**Comment by easypzlemonsqueezy** [comment]
it might have to do with how your individual events are being executed since your outputs are still in chronological order

**Comment by prisciliapangg** [comment]
I think the issue might be with polling twice. What I did was to add the string within the first while loop (such that output += [whatever event polled out]).

**Comment by nigel-tannn** [comment]
I don't think the issue is with the comparator, your output is sorted accurately according to time, it should be an issue with the simulator though, your customer 3 and 4 are being served before they have arrived.

**Comment by secretApple** [comment]
your timings look incorrect, may want to check if the timestamp is updated and server too
